File: babybites.hs
import Data.List
import Data.Maybe

main :: IO()
main = do
 contents <- getContents
 let count = words $ last $ lines contents
 let indexCount = zip [1..] count
 let result = find (\(i, saying) -> not (saying == "mumble") && not (i == (read :: String -> Int) saying)) indexCount
 if isNothing result
 then putStrLn "makes sense"
 else putStrLn "something is fishy"

File: babybites.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int n;
	cin >> n;
	string count;
	bool ok = true;
	for (int i = 1; i <= n; ++i) {
		cin >> count;

		if (ok) {
			if (count == "mumble")
				continue;

			int x = stoi(count);
			if (x != i) {
				ok = false;
				cout << "something is fishy";
			}	
		}
	}

	if (ok) cout << "makes sense";
}
